src/main/java/com/gluonhq/substrate/ProjectConfiguration.java

```diff
@@ -206,6 +206,20 @@ public void setCompilerArgs(List<String> compilerArgs) {
         this.compilerArgs = compilerArgs;
     }
 
+    public String getAppId() {
+        return appId;
+    }
+
+    /**
+     * Sets the AppID. This acts as the application identifier in various platforms.
+     * For Android, this is the equivalent of 'package' name of the application.
+     * 
+     * @param appId The application ID of the application.
+     */
+    public void setAppId(String appId) {
+        this.appId = appId;
```

Comment:
If this is a mandatory field (it can't be null), we should have a default value, or at least SubstrateDispatch should set it. 
Also, have you run tests?